commands/channel_test.go

```diff
@@ -2176,3 +2176,261 @@ func (s *MmctlUnitTestSuite) TestRenameChannelCmd() {
+
+	s.Run("should not create channel without name", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+
+		teamName := "teamName"
+		channelDisplayName := "channelDisplayName"
+		argsTeamChannel := teamName + ":" + channelDisplayName
+		args := []string{""}
+		args[0] = argsTeamChannel
+
+		cmd.Flags().String("team", teamName, "Team Name")
+		cmd.Flags().String("display_name", channelDisplayName, "Channel Display Name")
+
+		err := createChannelCmdF(s.client, cmd, args)
+		s.Require().EqualError(err, fmt.Sprintf("name is required"))
+	})
+
+	s.Run("should not create channel without team", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+
+		channelName := "channelName"
+		channelDisplayName := "channelDisplayName"
+		argsTeamChannel := channelName + ":" + channelDisplayName
+		args := []string{""}
+		args[0] = argsTeamChannel
+
+		cmd.Flags().String("name", channelName, "Channel Name")
+		cmd.Flags().String("display_name", channelDisplayName, "Channel Display Name")
+
+		err := createChannelCmdF(s.client, cmd, args)
+		s.Require().EqualError(err, fmt.Sprintf("team is required"))
+	})
+
+	s.Run("should fail when team does not exist", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+
+		teamName := "teamName"
+		channelName := "channelName"
+		channelDisplayName := "channelDisplayName"
+		argsTeamChannel := teamName + ":" + channelName + ":" + channelDisplayName
+		args := []string{""}
+		args[0] = argsTeamChannel
+
+		cmd.Flags().String("team", teamName, "Team Name")
+		cmd.Flags().String("name", channelName, "Channel Name")
+		cmd.Flags().String("display_name", channelDisplayName, "Channel Display Name")
+
+		mockError := &model.AppError{
+			Message: "mockError",
+		}
+
+		s.client.
+			EXPECT().
+			GetTeam(teamName, "").
+			Return(nil, &model.Response{Error: mockError}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetTeamByName(teamName, "").
+			Return(nil, &model.Response{Error: mockError}).
+			Times(1)
+
+		err := createChannelCmdF(s.client, cmd, args)
+		s.Require().EqualError(err, fmt.Sprintf("unable to find team: %s", teamName))
+	})
+
+	s.Run("should create public channel", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+
+		teamName := "teamName"
+		channelName := "channelName"
+		channelDisplayName := "channelDisplayName"
+		argsTeamChannel := teamName + ":" + channelName + ":" + channelDisplayName
+		args := []string{""}
+		args[0] = argsTeamChannel
+
+		cmd.Flags().String("team", teamName, "Team Name")
+		cmd.Flags().String("name", channelName, "Channel Name")
+		cmd.Flags().String("display_name", channelDisplayName, "Channel Display Name")
+
+		foundTeam := &model.Team{
+			Id:          "teamId",
+			Name:        teamName,
+			DisplayName: "teamDisplayName",
+		}
+
+		foundChannel := &model.Channel{
+			TeamId:      "teamId",
+			Name:        channelName,
+			DisplayName: channelDisplayName,
+			Header:      "",
+			Purpose:     "",
+			Type:        model.CHANNEL_OPEN,
+			CreatorId:   "",
```

Comment:
`Header`, `Purpose` and `CreatorId` will get zero value for a string which is already `""`. Therefore, we don't need to define like this.